addons/mail/static/tests/web/fields/m2x_avatar_user_tests.js

```diff
@@ -195,6 +196,127 @@ QUnit.test('many2many_avatar_user widget edited by the smart action "Assign to..
+
+            // Assign me (Luigi)
+            await triggerHotkey("alt+shift+i");
+            await contains(".o_field_many2one_avatar_user input", 1, { value: "Luigi" });
+
+            // Unassign me
+            await triggerHotkey("control+k");
+            await click("#o_command_2");
+            await contains(".o_field_many2one_avatar_user input", 1, { value: "" });
+        });
+    }
+);
+
+QUnit.test(
+    'many2one_avatar_user widget edited by the smart action "Assign to me" in list view',
+    async (assert) => {
+        const pyEnv = await startServer();
+        const [partnerId_1, partnerId_2] = pyEnv["res.partner"].create([
+            { name: "Mario" },
+            { name: "Luigi" },
+        ]);
+        const [userId_1, userId_2] = pyEnv["res.users"].create([
+            { name: "Mario", partner_id: partnerId_1 },
+            { name: "Luigi", partner_id: partnerId_2 },
+        ]);
+        pyEnv["m2x.avatar.user"].create([{ user_id: userId_2 }, { user_id: userId_1 }]);
+        const views = {
+            "m2x.avatar.user,false,list":
+                '<tree multi_edit="1"><field name="user_id" widget="many2one_avatar_user"/></tree>',
+        };
+        await pyEnv.withUser(userId_2, async () => {
+            const { openView } = await start({ serverData: { views } });
+            await openView({
+                type: "ir.actions.act_window",
+                target: "current",
+                res_model: "m2x.avatar.user",
+                view_mode: "list",
+                views: [[false, "list"]],
+            });
+            await contains(".o_field_many2one_avatar_user .o_form_uri span:eq(0)", {
+                text: "Luigi",
+            });
+            await contains(".o_field_many2one_avatar_user .o_form_uri span:eq(1)", {
+                text: "Mario",
+            });
+            // Select all
+            await click(".o_list_table > thead .o_list_controller input");
+            await triggerHotkey("control+k");
+            await contains(".o_command", { text: "Assign to meALT + SHIFT + I" });
+
+            // Assign me (Luigi)
+            await triggerHotkey("alt+shift+i");
+            // Multi-edit confirmation dialog
+            await contains(".o_dialog");
+            // Cancel
+            await click(".o_dialog .modal-footer button:nth-child(2)");
+            await contains(".o_field_many2one_avatar_user .o_form_uri span:eq(0)", {
+                text: "Luigi",
+            });
+            await contains(".o_field_many2one_avatar_user .o_form_uri span:eq(1)", {
+                text: "Mario",
+            });
+
+            // Assign me (Luigi)
+            await triggerHotkey("alt+shift+i");
+            // Multi-edit confirmation dialog
+            await contains(".o_dialog");
+            // Confirm
+            await click(".o_dialog .modal-footer button:nth-child(1)");
+            await nextTick();
```

Comment:
@seb-odoo do you know why it won't work without this single `nextTick()`?

Reply:
Hello, I would guess the following contains checks (l299 and 302) do not depend on the result of the click above (l297), so they immediately resolve, leading to a crash further in the test because it didn't wait long enough to proceed with the following actions (l307/l310) which leads to expected dialog (l312) not appearing because the conditions for it were not met.

I would suggest to make assertion l299 either more specific, or completely different, based on some change can be observed only following the click of l297, and not something that is already true before it.

Reply:
I see, thanks for the feedback :)